aws-logs-destination/src/main/java/software/amazon/logs/destination/CreateHandler.java

```diff
@@ -0,0 +1,55 @@
+package software.amazon.logs.destination;
+
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.cloudformation.Action;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    public static final String DESTINATION_CREATE_GRAPH = "AWS-Logs-Destination::Create";
+
+    public static final String DESTINATION_POLICY_CREATE_GRAPH = "AWS-Logs-DestinationPolicy::Create";
+
+    private static final String DEFAULT_LOGICAL_RESOURCE_IDENTIFIER = "";
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        // Verify if a destination is already present with same identifier
+        return ProgressEvent.progress(model, callbackContext)
+                .then(progress -> preCreateCheck(proxy, callbackContext, proxyClient, model).done((response) -> {
+                    if (isDestinationListNullOrEmpty(response)) {
+                        return ProgressEvent.progress(model, callbackContext);
+                    }
+                    return ProgressEvent.defaultFailureHandler(new CfnAlreadyExistsException(null),
+                            HandlerErrorCode.AlreadyExists);
+                }))
+                .then(progress -> putDestination(proxy, callbackContext, proxyClient, model, DESTINATION_CREATE_GRAPH,
+                        logger, Action.CREATE))
+                .then(progress -> putDestinationPolicy(proxy, callbackContext, proxyClient, model,
+                        DESTINATION_POLICY_CREATE_GRAPH, logger, Action.CREATE))
+                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient,
+                        logger));
+    }
+
+    private String generateDestinationId(ResourceHandlerRequest<ResourceModel> request) {
```

Comment:
where is it used and do you even need it if ur primary id a required property?

Reply:
Thank you for pointing out it is not being used will remove it